core.go

```diff
@@ -70,38 +72,32 @@ func (c *core) Check(ent zapcore.Entry, ce *zapcore.CheckedEntry) *zapcore.Check
 func (c *core) Write(ent zapcore.Entry, fs []zapcore.Field) error {
 	clone := c.with(fs)
 
-	/*
-		only when we have local sentryScope to avoid collecting all breadcrumbs ever in a global scope
-	 */
+	// only when we have local sentryScope to avoid collecting all breadcrumbs ever in a global scope
 	if c.cfg.EnableBreadcrumbs && c.cfg.BreadcrumbLevel.Enabled(ent.Level) && c.sentryScope != nil {
 		breadcrumb := sentry.Breadcrumb{
+			Message:   ent.Message,
 			Data:      clone.fields,
 			Level:     sentrySeverity(ent.Level),
-			Message:   ent.Message,
 			Timestamp: ent.Time,
-			Type:      "default",
 		}
-		c.sentryScope.AddBreadcrumb(&breadcrumb, maxLimit)
+
+		c.sentryScope.AddBreadcrumb(&breadcrumb, maxBreadcrumbs)
 	}
 
 	if ent.Level.Enabled(c.cfg.Level) {
 		event := sentry.NewEvent()
 		event.Message = ent.Message
 		event.Timestamp = ent.Time
 		event.Level = sentrySeverity(ent.Level)
-		event.Platform = "Golang"
 		event.Extra = clone.fields
 		event.Tags = c.cfg.Tags
+		event.Exception = clone.createExceptions()
 
-		if !c.cfg.DisableStacktrace {
-			trace := sentry.NewStacktrace()
-			if trace != nil {
-				trace.Frames = filterFrames(trace.Frames)
-				event.Exception = []sentry.Exception{{
-					Type:       ent.Message,
-					Value:      ent.Caller.TrimmedPath(),
-					Stacktrace: trace,
-				}}
+		if event.Exception == nil && !c.cfg.DisableStacktrace && c.client.Options().AttachStacktrace {
```

Comment:
we skip it for exceptions because it already contains stacktrace?

Reply:
Yes. I don't think it makes sense to create an artificial stack trace from "another place" when we already have one which is as exact as can be.